### PR TITLE
Refactor/code structure

### PR DESCRIPTION
**Pull request by nattallius**

### CODE REVIEW

File: src/plugins/settings/setting.ts [hunk 1]
```diff
@@ -13,20 +13,14 @@ import {WARNING_MSG} from '../../utils/warning-msg';
...
   @attr() public target: string;
 
-  public get settingContainer(): HTMLElement | null {
-    return this.closest(UIPSettings.is);
-  }
-
-  public get settings() {
-    return this.settingContainer as UIPSettings;
+  public get $settings() {
+    return this.closest(UIPSettings.is) as UIPSettings;
```
Comment (Sisha3342) [inline src/plugins/settings/setting.ts:23]: cosmetic: guess we should use memoize decorator here

**Review by Sisha3342 — APPROVED** [review]

**Review by AlexanderBazukevich — APPROVED** [review]

**Review by ala-n — APPROVED** [review]

File: src/settings/text-setting/text-setting.ts [hunk 2]
```diff
@@ -12,16 +13,18 @@ export class UIPTextSetting extends UIPSetting {
 
   /** Setting's visible name. */
   @attr({defaultValue: ''}) public label: string;
+
   /** Text input to change setting's value. */
-  protected $field: HTMLInputElement;
+  @memoize()
+  protected get $field() {
```
Comment (yadamskaya) [inline src/settings/text-setting/text-setting.ts:19]:
```suggestion
  protected get $field(): HTMLInputElement {
```

File: src/settings/bool-setting/bool-setting.ts [hunk 3]
```diff
@@ -27,16 +28,18 @@ export class UIPBoolSetting extends UIPSetting {
    * `append` - appending [attribute's]{@link UIPSetting#attribute} value to attribute's value.
    */
   @attr({defaultValue: 'replace'}) public mode: 'replace' | 'append';
+
   /** Checkbox field to change setting's value. */
-  protected $field: HTMLInputElement;
+  @memoize()
+  protected get $field() {
```
Comment (yadamskaya) [inline src/settings/bool-setting/bool-setting.ts:34]:
```suggestion
  protected get $field(): HTMLInputElement {
```

**Review by PolinaVolchetskaya — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

File: src/plugins/settings/setting.ts (hunk 1 above)
Comment (nattallius) [reply src/plugins/settings/setting.ts:25]: good catch, thank you, fixed!

**Pull request merged**